plugin/kubernetes/controller.go

```diff
@@ -130,16 +133,29 @@ func newdnsController(ctx context.Context, kubeClient kubernetes.Interface, opts
 	}
 
 	if opts.initEndpointsCache {
-		dns.epLister, dns.epController = object.NewIndexerInformer(
-			&cache.ListWatch{
-				ListFunc:  endpointsListFunc(ctx, dns.client, api.NamespaceAll, dns.selector),
-				WatchFunc: endpointsWatchFunc(ctx, dns.client, api.NamespaceAll, dns.selector),
-			},
-			&api.Endpoints{},
-			cache.ResourceEventHandlerFuncs{AddFunc: dns.Add, UpdateFunc: dns.Update, DeleteFunc: dns.Delete},
-			cache.Indexers{epNameNamespaceIndex: epNameNamespaceIndexFunc, epIPIndex: epIPIndexFunc},
-			object.DefaultProcessor(object.ToEndpoints(opts.skipAPIObjectsCleanup), dns.recordDNSProgrammingLatency),
-		)
+		if opts.useEndpointSlices {
```

Comment:
this needs to be simplified; seems `&api.EndpointSlice{}` is the only difference? That can be set latter in a small if-statement.

The endpoint slice also missed the dns latency for some reason?

Reply:
Thanks for reviewing! Agreed, I can reduce some code there.

> The endpoint slice also missed the dns latency for some reason?

When I tested in a local cluster, I saw that EndpointSlices did not have the annotation (or equivalent other field) required to calculate latency (something that holds the date of the last update).  I'll double check in latest k8s release.

Reply:
Ah, I found an updated timestamp in another place in the EndpointSlice object, I'll adjust the code to use it.

Reply:
listWatch can be moved out of the if-then, and just assigned, making this more readeble